tests/test_app.py

```diff
@@ -68,13 +109,74 @@ def test_get_property_invalid_path_404(self, mock_response, mock_get):
         response = self.app.get('/properties/%s' % street_address_url)
         assert response.status_code == 404
 
+    @mock.patch('service.server.get_property_address', return_value=one_DB_result)
+    @mock.patch('requests.post')
+    def test_get_property_address_called_correctly(self, mock_post, mock_get_property_address):
+        search_query = "PL6%208RU/PATTINSON%20DRIVE_100"
+
+        query_dict = {
+            'postcode': 'PL6 8RU',
+            'street': 'PATTINSON DRIVE',
+            'paon': '100',
+        }
+
+        response = self.app.get('/properties/%s' % search_query)
+
+        args, kwargs = mock_get_property_address.call_args
+
+        self.assertEqual(args, (query_dict, ))
+        self.assertEqual(kwargs, {})
+
+    @mock.patch('service.server.AddressBase.query.filter_by')
+    @mock.patch('service.server.AddressBase.query', return_value=FakeQuery(one_DB_result))
+    def test_get_property_address_called_correctly(self, mock_query, mock_filter_by):
+        query_dict = {
+            'postcode': 'PL6 8RU',
+            'street': 'PATTINSON DRIVE',
+            'paon': '100',
+        }
+
+        get_property_address(query_dict)
+        mock_filter_by.assert_has_calls(call(postcode=query_dict['postcode']))
+        mock_filter_by.assert_has_calls(call(thoroughfareName=query_dict['street']))
+
+        # TODO: is there a nicer way to do this?
```

Comment:
can't think of one right now :( Shall we remove the TODO? Is is something you'd like to address at some later point?